test/lit/passes/once-reduction.wast

```diff
@@ -0,0 +1,1425 @@
+    (drop (i32.const 100))
+  )
+
+  ;; CHECK:      (func $caller-if-1
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:   (block $block
+  ;; CHECK-NEXT:    (call $once)
+  ;; CHECK-NEXT:    (nop)
+  ;; CHECK-NEXT:    (nop)
+  ;; CHECK-NEXT:    (nop)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $once)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller-if-1
+    ;; Add more calls, and ones that are conditional.
+    (if
+      (i32.const 1)
+      (block
+        (call $once)
+        (call $once)
+        (call $once)
+        (call $once)
+      )
+    )
+    (call $once)
+    (call $once)
+  )
+
+  ;; CHECK:      (func $caller-if-2
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:   (call $once)
+  ;; CHECK-NEXT:   (block $block
+  ;; CHECK-NEXT:    (call $once)
+  ;; CHECK-NEXT:    (nop)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $once)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller-if-2
+    ;; Call in both arms. As we only handle dominance, and not merges, the first
+    ;; call after the if is *not* optimized.
+    (if
+      (i32.const 1)
+      (call $once)
+      (block
+        (call $once)
+        (call $once)
+      )
+    )
+    (call $once)
+    (call $once)
+  )
+
+  ;; CHECK:      (func $caller-loop-1
+  ;; CHECK-NEXT:  (loop $loop
+  ;; CHECK-NEXT:   (if
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:    (call $once)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (call $once)
+  ;; CHECK-NEXT:   (nop)
+  ;; CHECK-NEXT:   (br_if $loop
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller-loop-1
+    ;; Add calls in a loop.
+    (loop $loop
+      (if
+        (i32.const 1)
+        (call $once)
+      )
+      (call $once)
+      (call $once)
+      (br_if $loop (i32.const 1))
+    )
+    (call $once)
+    (call $once)
+  )
+
+  ;; CHECK:      (func $caller-loop-2
+  ;; CHECK-NEXT:  (loop $loop
+  ;; CHECK-NEXT:   (if
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:    (call $once)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (br_if $loop
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $once)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller-loop-2
+    ;; Add a single conditional call in a loop.
+    (loop $loop
+      (if
+        (i32.const 1)
+        (call $once)
+      )
+      (br_if $loop (i32.const 1))
+    )
+    (call $once)
+    (call $once)
+  )
+
+  ;; CHECK:      (func $caller-single
+  ;; CHECK-NEXT:  (call $once)
+  ;; CHECK-NEXT: )
+  (func $caller-single
+    ;; A short function with a single call.
+    (call $once)
+  )
+
+  ;; CHECK:      (func $caller-empty
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $caller-empty
+    ;; A tiny function with nothing at all.
+  )
```

Comment:
What is this function for? It doesn't contain `once` and it is not called from anywhere...

Reply:
It just tests that we do not crash when processing such a tiny function. (We used to at some point in testing...) I'll add a comment.